fsharp-backend/tests/TestUtils/TestUtils.fs

```diff
@@ -298,7 +298,7 @@ let executionStateFor
                 // This should be greater than zero, but there's confusion between
                 // different kinds of exceptions. Once we untangle handleException
                 // this can be re-enabled.
-                // FSTODO tc.exceptionReports.Length > 0
+                // FSTODO tc.exceptionReports.Length > 0 WHATISTHIS
```

Comment:
What's the same expectation here?
Has `handleException` been "untangled"?

Reply:
Yes it has actually! See here for the new logic https://github.com/darklang/dark/blob/63057e05434fb26134ace13693443c69111a0e76/fsharp-backend/src/LibExecution/Interpreter.fs#L680


I _think_ but am not sure that you could remove the `if isDError` here.

Any exception except CodeException and GrandUserException should indicate that the code is misbehaving. We may have to triage those misbehaviours individually, but the general idea is that other exceptions should be caught and handled closer to the callsite (and if they still want to throw exceptions, they should use a CodeException).